ansible/roles_ocp_workloads/ocp4_workload_rhtr_xraylab/tasks/storage.yaml

```diff
@@ -0,0 +1,154 @@
+  k8s_info:
+    api_version: "noobaa.io/v1alpha1"
+    kind: NooBaa
+    namespace: "{{ ocp4_workload_rhtr_xraylab_ocs_namespace }}"
+  register: noobaa_system_status
+  retries: 60
+  delay: 10
+  until: noobaa_system_status.resources|length > 0 and
+         noobaa_system_status.resources[0].status is defined and 
+         noobaa_system_status.resources[0].status.phase == 'Ready'
+  ignore_errors: true
+
+- when: ocp4_workload_rhtr_xraylab_ocs_install_mcg
+  block:
+  - when: ocp4_workload_rhtr_xraylab_ocs_mcg_pv_pool
+    block:
+    - name: "Create PV Pool BackingStore"
+      k8s:
+        state: "{{ state }}"
+        definition: "{{ lookup('template', 'storage/pv_pool_backingstore.yaml.j2') }}"
+
+    - name: "Create PV Pool BucketClass"
+      k8s:
+        state: "{{ state }}"
+        definition: "{{ lookup('template', 'storage/pv_pool_bucketclass.yaml.j2') }}"
+
+    - name: "Create PV Pool ObjectBucketClaim"
+      k8s:
+        state: "{{ state }}"
+        definition: "{{ lookup('template', 'storage/pv_pool_objectbucketclaim.yaml.j2') }}"
+
+    - name: "Wait for Bucket to exist"
+      when: not ocp4_workload_rhtr_xraylab_ocs_operator_workload_destroy|bool
+      k8s_info:
+        api_version: "objectbucket.io/v1alpha1"
+        kind: ObjectBucketClaim
+        name: "{{ ocp4_workload_rhtr_xraylab_ocs_mcg_pv_pool_bucket_name }}"
+        namespace: "{{ ocp4_workload_rhtr_xraylab_ocs_namespace }}"
+      register: bucket
+      until: (bucket.resources|length) > 0
+      retries: 30
+      delay: 10
+
+    - name: "Wait for Bucket to have status"
+      when: not ocp4_workload_rhtr_xraylab_ocs_operator_workload_destroy|bool
+      k8s_info:
+        api_version: "objectbucket.io/v1alpha1"
+        kind: ObjectBucketClaim
+        name: "{{ ocp4_workload_rhtr_xraylab_ocs_mcg_pv_pool_bucket_name }}"
+        namespace: "{{ ocp4_workload_rhtr_xraylab_ocs_namespace }}"
+      register: bucket
+      until: (bucket.resources|first).status is defined
+      retries: 30
+      delay: 10
+
+    - name: "Wait for Bucket to become bound"
+      when: not ocp4_workload_rhtr_xraylab_ocs_operator_workload_destroy|bool
+      k8s_info:
+        api_version: "objectbucket.io/v1alpha1"
+        kind: ObjectBucketClaim
+        name: "{{ ocp4_workload_rhtr_xraylab_ocs_mcg_pv_pool_bucket_name }}"
+        namespace: "{{ ocp4_workload_rhtr_xraylab_ocs_namespace }}"
+      register: bucket
+      until: (bucket.resources|first).status.phase == "Bound"
+      retries: 75
+      delay: 10
+
+- name: "Create ObjectStore"
+  when: ocp4_workload_rhtr_xraylab_ocs_install_rgw
+  k8s:
+    state: "{{ state }}"
+    definition: "{{ lookup('template', 'storage/rgw_objectstore.yaml.j2') }}"
+
+- name: "Deploy Ceph Toolbox"
+  when: ocp4_workload_rhtr_xraylab_ocs_install_rgw
+  shell: |
+    oc patch OCSInitialization ocsinit -n openshift-storage --type json --patch  '[{ "op": "replace", "path": "/spec/enableCephTools", "value": true }]'
```

Comment:
Ha. I just learned about that on Monday. :-) 

You *should* be able to use k8s with "merge" strategy for that as well. Basically just have the YAML of what you want changed.

Reply:
OK, will try that.

Reply:
Works now with the k8s module. I did not know this would work too as patching CRDs as always been tricky. Normally it's a json strategy, not a merge in the sense of the patch command, but it seems that k8s is handling it well.